packages/core/src/lib/waku_peer_exchange/index.ts

```diff
@@ -63,13 +63,26 @@ class WakuPeerExchange implements PeerExchange {
         bytes.append(chunk);
       });
 
-      const response = PeerExchangeRPC.decode(bytes).response;
+      const decoded = PeerExchangeRPC.decode(bytes).response;
 
-      if (!response) {
+      if (!decoded) {
         throw new Error("Failed to decode response");
       }
 
-      return response;
+      const enrPromises: Promise<ENR>[] = [];
+      for (const peerInfo of decoded.peerInfos) {
+        if (!peerInfo.ENR) continue;
+        const enr = ENR.decode(peerInfo.ENR);
+        enrPromises.push(enr);
+      }
+
+      const peerInfos = (await Promise.all(enrPromises)).map((enr) => {
+        return {
+          ENR: enr,
+        };
+      });
+
+      return { peerInfos };
```

Comment:
If you want to consolidate the processing (question of style , just using this opportunity to show something cool I learn in the past year) you can use a type guard

1. create a type guard

```
    const isDefined = (
      enr: ENR | undefined
    ): enr is ENR => {
      return !!enr
    };
```

Use it with filter:

```suggestion
			const enrs = await Promise.all(
								decoded.peerInfos.map(async (peerInfo) => {
										if (!peerInfo.ENR) return;
										const enr = await ENR.decode(peerInfo.ENR)
										return { ENR: enr	} 
								}
			)
			
      return enrs.filter(isDefined);
```

Reply:
Cool concept, thanks!

How are we benefitting from using it here?

Reply:
> How are we benefitting from using it here? All ENRs here are implicitly defined correctly.

They are not because of ` if (!peerInfo.ENR) continue;`

If you were to use this statement in `map` then it would need to be a `return` which is the same as `return undefined`.